nifi-extension-bundles/nifi-amqp-bundle/nifi-amqp-processors/src/main/java/org/apache/nifi/amqp/processors/PublishAMQP.java

```diff
@@ -268,4 +357,43 @@ private Map<String, Object> validateAMQPHeaderProperty(String amqpPropValue, Cha
         }
         return headers;
     }
+    public enum InputHeaderSource implements DescribedValue {
+
+        ATTRIBUTES("headersFromAttributes", "Attributes Matching Regex",
+            "Select attributes based on regex pattern to put in rabbitmq headers. Key of the attribute will be used as header key"),
+        STRING("headersFromString", "Attribute 'amp$headers' Value",
+            "Prepare headers from 'amp$headers' attribute string"),
+        BOTH("headersFromBoth", "Regex Match And 'amp$headers' Value",
```

Comment:
The value of `BOTH` is confusing and would not translate well if some new value were introduced. It seems like it should be removed, along with the Precedence property.